backend/src/main/java/com/pickpick/slackevent/application/message/MessageThreadBroadcastService.java

```diff
@@ -0,0 +1,109 @@
+package com.pickpick.slackevent.application.message;
+
+import com.pickpick.channel.domain.Channel;
+import com.pickpick.channel.domain.ChannelRepository;
+import com.pickpick.exception.SlackApiCallException;
+import com.pickpick.exception.member.MemberNotFoundException;
+import com.pickpick.member.domain.Member;
+import com.pickpick.member.domain.MemberRepository;
+import com.pickpick.message.domain.MessageRepository;
+import com.pickpick.slackevent.application.SlackEvent;
+import com.pickpick.slackevent.application.SlackEventService;
+import com.pickpick.slackevent.application.message.dto.SlackMessageDto;
+import com.slack.api.methods.MethodsClient;
+import com.slack.api.methods.SlackApiException;
+import com.slack.api.model.Conversation;
+import java.io.IOException;
+import java.util.Map;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Transactional
+@Service
+public class MessageThreadBroadcastService implements SlackEventService {
+
+    private static final String EVENT = "event";
+    private static final String USER = "user";
+    private static final String TIMESTAMP = "ts";
+    private static final String TEXT = "text";
+    private static final String CLIENT_MSG_ID = "client_msg_id";
+    private static final String CHANNEL = "channel";
+    private final MessageRepository messages;
+    private final MemberRepository members;
+    private final ChannelRepository channels;
+    private final MethodsClient slackClient;
+
+    public MessageThreadBroadcastService(final MessageRepository messages, final MemberRepository members,
+                                         final ChannelRepository channels, final MethodsClient slackClient) {
+        this.messages = messages;
+        this.members = members;
+        this.channels = channels;
+        this.slackClient = slackClient;
+    }
+
+    @Override
+    public void execute(final Map<String, Object> requestBody) {
+        SlackMessageDto slackMessageDto = convert(requestBody);
+
+        save(slackMessageDto);
+    }
+
+    public void saveWhenSubtypeIsMessageChanged(final SlackMessageDto slackMessageDto) {
+        if (shouldSave(slackMessageDto)) {
+            save(slackMessageDto);
+        }
+    }
+
+    private boolean shouldSave(final SlackMessageDto slackMessageDto) {
+        return messages.findBySlackId(slackMessageDto.getSlackId()).isEmpty();
+    }
+
+    private void save(final SlackMessageDto slackMessageDto) {
+        String memberSlackId = slackMessageDto.getMemberSlackId();
+        Member member = members.findBySlackId(memberSlackId)
+                .orElseThrow(() -> new MemberNotFoundException(memberSlackId));
+
+        String channelSlackId = slackMessageDto.getChannelSlackId();
+
+        Channel channel = channels.findBySlackId(channelSlackId)
+                .orElseGet(() -> createChannel(channelSlackId));
```

Comment:
인라인 처리 별로일까요?

Reply:
```suggestion
        Member member = members.findBySlackId(slackMessageDto.getMemberSlackId())
                .orElseThrow(() -> new MemberNotFoundException(slackMessageDto.getMemberSlackId()));

        Channel channel = channels.findBySlackId(slackMessageDto.getChannelSlackId())
                .orElseGet(() -> createChannel(slackMessageDto.getChannelSlackId()));
```

인라인 처리하면 findBySlackId랑 exception 두 곳에서 모두 사용되어서 전 현재 상태가 좋다고 생각해요!
이건 연로그랑 리차드 의견도 들어보고싶습니다 :)

Reply:
저는 인라인을 하지 않는 현재를 선호합니다~

Reply:
저도 지금이 좋은 것 같아요~😄